cps/main.c

```diff
@@ -676,6 +676,35 @@ assign_cps_queue_ids(struct cps_config *cps_conf)
 	return ret;
 }
 
+/*
+ * In order to avoid a race condition in which the interface @if_name is
+ * seen as not available in the kernel, wait for its interface index
+ * to be assigned.
+ */
+static int
+wait_for_the_kernel(const char *if_name, const unsigned int wait_trials)
+{
+	unsigned int i = 0;
+
+	do {
+		if (if_nametoindex(if_name) > 0) {
+			/* Interface is available. */
+			return 0;
+		}
+
+		i++;
+		sleep(1);
+	/*
+	 * Use greater than in the test below to account for the fact that
+	 * there is no wait before the first call of if_nametoindex().
+	 */
+	} while (i > wait_trials);
```

Comment:
If `wait_trials` is greater than `0`, then this condition will always be `false` after the first call of `if_nametoindex()`.